controllers/argocd_metrics_controller.go

```diff
@@ -292,6 +308,79 @@ func (r *ArgoCDMetricsReconciler) createPrometheusRuleIfAbsent(namespace string,
 	return err
 }
 
+func (r *ArgoCDMetricsReconciler) createDashboardsIfAbsent(reqLogger logr.Logger) error {
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: dashboardNamespace}, &corev1.Namespace{})
+	if err != nil {
+		reqLogger.Info("OpenShift dashboards are not installed, skipping dashboard installation",
+			"Namespace", dashboardNamespace)
+		return nil
+	}
+
+	entries, err := dashboards.ReadDir(dashboardFolder)
+	if err != nil {
+		reqLogger.Error(err, "Could not read list of embedded dashboards")
+		return err
+	}
+
+	for _, entry := range entries {
+		reqLogger.Info("Processing dashboard", "Namespace", dashboardNamespace, "Name", entry.Name())
+
+		if !entry.IsDir() {
+			dashboard, err := newDashboard(entry.Name(), dashboardNamespace)
+			if err != nil {
+				reqLogger.Info("There was an error creating dashboard ", "Namespace", dashboardNamespace, "Name", entry.Name())
+				return err
+			}
+
+			existingDashboard := &corev1.ConfigMap{}
+
+			err = r.Client.Get(context.TODO(), types.NamespacedName{Name: dashboard.Name, Namespace: dashboardNamespace}, existingDashboard)
+			if err == nil {
+				reqLogger.Info("A dashboard instance already exists",
+					"Namespace", existingDashboard.Namespace, "Name", existingDashboard.Name)
+				continue
+			}
+			if errors.IsNotFound(err) {
+				reqLogger.Info("Creating new dashboard",
+					"Namespace", dashboard.Namespace, "Name", dashboard.Name)
+				err := r.Client.Create(context.TODO(), dashboard)
+				if err != nil {
+					reqLogger.Error(err, "Error creating a new dashboard",
+						"Namespace", dashboard.Namespace, "Name", dashboard.Name)
+					// Should we return an error or continue with other dashboards?
+					return err
+				}
+			}
+		}
+	}
+	return nil
+}
+
+func newDashboard(filename string, namespace string) (*corev1.ConfigMap, error) {
```

Comment:
can we call this `newDashboardConfigMap` instead? Makes it easier to follow the code knowing what the function is actually doing without getting into it